Lib/py_compile.py

```diff
@@ -137,6 +137,10 @@ def compile(file, cfile=None, dfile=None, doraise=False, optimize=-1):
     except FileExistsError:
         pass
     source_stats = loader.path_stats(file)
+    sde = os.environ.get('SOURCE_DATE_EPOCH')
+    if sde and source_stats['mtime'] > int(sde):
```

Comment:
I concur for the unit test. Document the variable in Doc/library/py_compile.rst

What happens if SOURCE_DATE_EPOCH is not a valid integer? Should it be ignored or raise a nice error message?

Reply:
You might convert sde to int to raise a nice error message if the conversion fails.

Reply:
I addressed 4 of 5 comments, but for this one I don't know how you would prefer it to be...

Reply:
also, if `SOURCE_DATE_EPOCH` is set, it is very uncommon (aka exceptional) to be a non-int, because it is set by programs rather than users most of the time.

Reply:
Python should emulate what other compilers do, for example https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=e3e8c48c4a494d9da741c1c8ea6c4c0b7c4ff934

Also see https://reproducible-builds.org/specs/source-date-epoch/#idm53
> If the value is malformed, the build process SHOULD exit with a non-zero error code.

Reply:
I say convert the string value to an int in a separate step in a `try`, catch the `ValueError`, and raise a chained `ValueError` that `SOURCE_DATE_EPOCH` is malformed.